provisioning/device/samples/register_x509.js

```diff
@@ -35,6 +39,22 @@ deviceClient.register(function(err, result) {
     console.log('registration succeeded');
     console.log('assigned hub=' + result.assignedHub);
     console.log('deviceId=' + result.deviceId);
+    var connectionString = 'HostName=' + result.assignedHub + ';DeviceId=' + result.deviceId + ';x509=true';
+    var hubClient = Client.fromConnectionString(connectionString, iotHubTransport);
+    hubClient.setOptions(deviceCert);
+    hubClient.open(function(err) {
+      if (err) {
+        console.error('Could not connect: ' + err.message);
+      } else {
+        console.log('Client connected');
+        var message = new Message('Hello world');
+        hubClient.sendEvent(message, function(err, res) {
+          if (err) console.log('send error: ' + err.toString());
+          if (res) console.log('send status: ' + res.constructor.name);
+          process.exit(1);
+        });
+      }
+    });
```

Comment:
just something I want to understand, I remember when Rido looked at this, we recommended using the X509AuthenticationProvider... I see we're not using it here-- how does this component fit in in general?